pallets/capacity/src/lib.rs

```diff
@@ -834,9 +834,12 @@ impl<T: Config> Pallet<T> {
 	) -> BalanceOf<T> {
 		let account_balance =
 			T::Currency::reducible_balance(&staker, Preservation::Preserve, Fortitude::Polite);
-		account_balance
-			.saturating_sub(T::MinimumTokenBalance::get())
-			.min(proposed_amount)
+		let stakable_amount = account_balance.saturating_sub(T::MinimumTokenBalance::get());
+		if stakable_amount >= proposed_amount {
+			proposed_amount
+		} else {
+			Zero::zero()
+		}
```

Comment:
At the calling end of this function a validation is to check if returned amount > 0 and a error is thrown then